### DISCUSSION TITLE
Poor Pallet Segmentation with Fine-Tuned YOLOv8 on Ouster LiDAR 2D Image

### DISCUSSION THREAD

**Discussion by Hemanth-mandava-neuropi** [ultralytics/ultralytics#21431] [Q&A]
Hello Ultralytics team,

I'm working on a project involving YOLO segmentation using 2D images derived from an Ouster LiDAR sensor. These images include reflectivity (0–255 grayscale) and near-infrared (NIR) data. Here's a summary of my workflow and the issue I'm facing:

Project Background:
I initially tested the pre-trained YOLOv8m segmentation model (RGB-trained) on my LiDAR-generated 2D images (reflectivity/NIR).

The model successfully detected cars and people on my test site with no fine-tuning.

I then fine-tuned the YOLOv8m model using RGB images of pallets via transfer learning.

When I tested the fine-tuned model on LiDAR images of pallets, the detection and segmentation results were poor.

To double-check, I also tested the Ultralytics pre-trained person segmentation model on my LiDAR data—it performed perfectly, detecting people accurately even in LiDAR-generated images.

My Assumptions:
Pallets occupy a small pixel area in the 2D LiDAR images, making them hard to detect. In contrast, people appear larger and retain more distinguishable features.

There is a sensor mounting bracket visible in every LiDAR frame, which may be confusing the model (see attached image). This may interfere with pallet detection.

The images exhibit banding and striping artifacts, potentially degrading features and causing further confusion—although these do not seem to affect person detection.

Questions:
Would it help to fine-tune the model with annotated images including both pallets and sensor mounting, so the model learns to ignore or distinguish the mounting bracket during inference?

Do you have any recommendations to improve segmentation for small objects like pallets in LiDAR images?

Is it advisable to train directly on reflectivity/NIR images, or should I try converting them to pseudo-RGB before training?

Any guidance on how to resolve this issue or improve the model's robustness to LiDAR image artifacts would be greatly appreciated.
note the dimensions of images used for inference is 1024,512  * 128
<img width="386" height="93" alt="Screenshot 2025-07-02 145412" src="https://github.com/user-attachments/assets/01ee12c0-72aa-4d3f-a131-ea45182865c8" />
<img width="766" height="95" alt="reflectivitysegimage" src="https://github.com/user-attachments/assets/8830b008-0f74-4b65-90ed-75f335c24de4" />
<img width="383" height="92" alt="ref pallet" src="https://github.com/user-attachments/assets/49874e88-495f-4796-b443-2b1dc26bdd44" />
<img width="196" height="50" alt="personseg" src="https://github.com/user-attachments/assets/f5e59a37-c4a0-450f-835a-2e3a399c0035" />

Thank you!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Hemanth-mandava-neuropi, thank you for sharing your detailed workflow and observations with Ultralytics 🚀! For new users or anyone looking for reference, we recommend checking out the [Docs](https://docs.ultralytics.com/), which offer a wealth of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples. Many common questions are addressed there.

If this is a 🐛 Bug Report, please include a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further.

For custom training or segmentation ❓ questions like yours, please provide as much detail as possible—including dataset samples and training logs—and make sure you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Feel free to join our community to discuss your project with others:
- For real-time support, visit our [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, head to [Discourse](https://community.ultralytics.com/)
- For sharing and exploring ideas, our [Subreddit](https://www.reddit.com/r/Ultralytics/) is also available

## Upgrade

To ensure your issue is not already resolved, please upgrade to the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of the following up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

🔔 This is an automated response to help you get started, but an Ultralytics engineer will review your discussion and assist you further soon!

**Reply by glenn-jocher** [reply]
Great question about domain adaptation challenges! Your analysis is spot-on - training on RGB then inferring on LiDAR reflectivity/NIR creates a significant domain gap.

For your specific questions: **Yes, absolutely include the sensor mounting bracket in your training annotations** so the model learns to ignore it. For small objects like pallets, try [image tiling](https://docs.ultralytics.com/guides/model-evaluation-insights/#image-tiling-for-small-objects) by splitting your 1024x512 images into smaller segments to maintain resolution while improving detection accuracy.

**Train directly on reflectivity/NIR data** rather than converting to pseudo-RGB - this preserves the unique characteristics of your sensor data. Consider creating a mixed dataset with both RGB pallet images and LiDAR reflectivity images of pallets to bridge the domain gap during training.

The key issue is that your model learned pallet features from RGB color/texture but LiDAR images rely on reflectivity patterns. You'll likely need LiDAR-specific training data for optimal performance, especially given the banding artifacts you mentioned.

**Reply by Hemanth-mandava-neuropi** [reply]
@glenn-jocher  Apologies for the back‑and‑forth—there was a typo in my last message. My LiDAR 2D images are actually 1024×128 and 512×128 (not 1024×512), and I also tried your suggested 128×128 tiling strategy, but the results were still poor.what strategies you recommend or i can do research improve pallet detection in wide and short frames. do i need to implement sahi?
2. actually the Lidar images of resolution 512 * 128 and RGB images of 640 * 640,Is it advisable to train on such a mixed‑resolution dataset? If so, what’s the best way to handle the differing image sizes.currently i fine tuned model with rgb pallet images can i train same model with Lidar images of 512 * 128 like curriculum learning. does it recommendable way of doing?

**Reply by glenn-jocher** [reply]
@Hemanth-mandava-neuropi Thanks for the clarification on dimensions! For those wide, short frames (512×128), **SAHI would definitely help** - it's designed exactly for this type of challenging aspect ratio where objects might be missed due to the unusual frame dimensions.

**Mixed-resolution training is perfectly fine** - YOLO automatically handles different input sizes during training by resizing images to the model's input size (typically 640×640). The model learns to work with various aspect ratios internally.

However, **instead of curriculum learning (RGB first, then LiDAR), try joint training from the start** with both RGB and LiDAR images mixed together. This helps the model learn shared features between domains rather than potentially "forgetting" RGB patterns when you add LiDAR data later.

Since your LiDAR images are so wide and short, also consider training with a custom `imgsz` parameter that better matches your aspect ratio, like `imgsz=512` or experiment with rectangular training shapes to preserve more of the original image structure.
